scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -602,7 +604,8 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
                 } else
                   // always add indent for indented `match` block
                   // check the previous token to avoid infinity loop
-                  prev.is[KwMatch] && next.is[KwCase] && token.isNot[Indentation.Indent]
+                  (!prev.prev.is[soft.KwEnd] && prev.is[KwMatch]) &&
+                  next.is[KwCase] && token.isNot[Indentation.Indent]
```

Comment:
sorry for being a pain but this is something that i learned the hard way in scalafmt. does prev refer to the previous token, skipping both whitespace and comments?

will this handle something like that:
```
    foo match // blah
    case bar => ...
    end match // blah
  case qux => ...
```

Reply:
It skips them:
```scala
    def prev: Token = {
      val prev = scannerTokens.apply(Math.max(token.index - 1, 0))
      if (prev.is[Whitespace] || prev.is[Comment]) prev.prev
      else prev
    }
```

Reply:
what about this:
```
val end = ...
end match
case theend => ...
end match
```

Reply:
this doesn't compile in Dotty from what I see.

Reply:
why not? `end` is not restricted https://scalacenter.github.io/scala-3-migration-guide/docs/incompatibilities/syntactic-changes.html#restricted-keywords

Reply:
Might be a bug? Or really there is a number of scenarios, where soft keywords don't work, but it's not stated.

Same with:
```scala
def extension(a : Int) = a + 2
extension(2)
```
will not work.

You need to do:
```scala
def extension(a : Int) = a + 2
`extension`(2)
```

So soft keywords are kind of dodgy and when I asked Martin he said that this is fine.

Reply:
https://github.com/lampepfl/dotty/issues/10076 for some context (I kind of argued with him there :sweat_smile: )